setup_utils.py

```diff
@@ -0,0 +1,149 @@
+import ast
+import os
+import re
+import subprocess
+
+
+_R_RC = re.compile("rc(\d+)$")
+
+
+def parse_version(path):
+    with open(path) as fp:
+        return _AssignmentParser().parse(fp.read())["version"]
+
+
+def write_version_py(filename, major, minor, micro, is_released,
+                     previous_version=None):
```

Comment:
The idea is that from time to time, you "increment" the starting point for computing build numbers. By default, we start from the first commit, but later, we may start from tag `v0.2.0` for `v0.3.0`, etc...